server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexTemplateService.java

```diff
@@ -240,19 +238,15 @@ private static void validateAndAddTemplate(final PutRequest request, IndexTempla
             templateBuilder.patterns(request.indexPatterns);
             templateBuilder.settings(request.settings);
 
-            Map<String, Map<String, Object>> mappingsForValidation = new HashMap<>();
-            for (Map.Entry<String, String> entry : request.mappings.entrySet()) {
+            if (request.mappings != null) {
                 try {
-                    templateBuilder.putMapping(entry.getKey(), entry.getValue());
-                } catch (Exception e) {
-                    throw new MapperParsingException("Failed to parse mapping [{}]: {}", e, entry.getKey(), e.getMessage());
+                    templateBuilder.putMapping(MapperService.SINGLE_MAPPING_NAME, request.mappings);
```

Comment:
This seems to both perform the refactor but also adjust the scope of the `try ... catch`. Is there a good reason to change this? Otherwise we could leave it as-is to minimize extra changes.

Reply:
This was to make `MetaDataIndexTemplateServiceTests#testBrokenMappings` pass - the refactoring moved around where things were being parsed, so it didn't get the exception that it was expecting.  However, looking at things more carefully, parsing happens when the initial `PutIndexTemplateRequest` is constructed, so this `try-catch` is entirely unnecessary.  We already test for broken mappings in `PutIndexTemplateRequestTests`, so I've removed the `try-catch` and the failing test.

Reply:
I'm not clear on why we need to remove the try/ catch and remove the test? I think the try/ catch is present to try to always throw a `MapperParsingException`, even in case the `templateBuilder.putMapping` call fails and throws a `NotXContentException`.

I tried just restoring the original structure below, and `testBrokenMappings` then passes:

```
if (request.mappings != null) {
                try {
                    templateBuilder.putMapping(MapperService.SINGLE_MAPPING_NAME, request.mappings);
                } catch (Exception e) {
                    throw new MapperParsingException("Failed to parse mapping [{}]: {}", e, "_doc", e.getMessage());
                }
                dummyIndexService.mapperService().merge(MapperService.SINGLE_MAPPING_NAME,
                    MapperService.parseMapping(xContentRegistry, request.mappings), MergeReason.MAPPING_UPDATE);
            }
```

It just seems nice to avoid making changes not related to the PR goal (and I don't have a full picture of our template validation strategy, so I'm not as confident as to whether we can remove the try/ catch and test).